pkg/sqlcmd/sqlcmd.go

```diff
@@ -327,24 +327,26 @@ func (s *Sqlcmd) IncludeFile(path string, processAll bool) error {
 	b := s.batch.batchline
 	utf16bom := unicode.BOMOverride(unicode.UTF8.NewDecoder())
 	unicodeReader := transform.NewReader(f, utf16bom)
-	scanner := bufio.NewScanner(unicodeReader)
-	buf := make([]byte, maxLineBuffer)
-	scanner.Buffer(buf, maxLineBuffer)
+	scanner := bufio.NewReader(unicodeReader)
 	curLine := s.batch.read
 	echoFileLines := s.echoFileLines
 	s.batch.read = func() (string, error) {
-		if !scanner.Scan() {
-			err := scanner.Err()
-			if err == nil {
-				return "", io.EOF
-			}
-			return "", err
+		var (
+			isPrefix bool  = true
+			err      error = nil
+			line, ln []byte
```

Comment:
> ln  [](http://example.com/codeflow?start=8&length=4)

i will experiment with pre-allocating ln as a property of s so we don't have to perform so many allocations.
There's no race condition so we should be able to reuse it.
The question becomes - is there a line length where we wouldn't want to keep the buffer allocated at that size? EG if we hit some line size like 1 GB, is it ok to keep a 1GB buffer around for the remainder of the process lifetime?

Reply:
You are freeing up the buffer for reuse anyway. I think the garbage collector would decide if it should be around for the process lifetime.

Reply:
setting the reference to `[:0]` doesn't free the buffer for GC, it just sets the length to 0 without reducing capacity.
The current code keeps the buffer around for the lifetime of the file read then lets it GC.